libs/execution_context/ebpf_program.c

```
@@ -437,29 +439,39 @@ ebpf_program_load_byte_code(ebpf_program_t* program, ebpf_instruction_t* instruc
 void
 ebpf_program_invoke(_In_ const ebpf_program_t* program, _In_ void* context, _Out_ uint32_t* result)
 {
+    uint64_t count = 0;
+    const ebpf_program_t* current_program = program;
     if (!program || program->program_invalidated) {
         *result = 0;
         return;
     }
 
-    if (program->parameters.code_type == EBPF_CODE_NATIVE) {
-        ebpf_program_entry_point_t function_pointer;
-        function_pointer = (ebpf_program_entry_point_t)(program->code_or_vm.code.code_pointer);
-        *result = (function_pointer)(context);
-    } else {
-        uint64_t out_value;
-        int ret = (uint32_t)(ubpf_exec(program->code_or_vm.vm, context, 1024, &out_value));
-        if (ret < 0) {
-            *result = ret;
+    for (count = 0; count < MAX_TAIL_CALL_CNT; count++) {
```

Should return an error if count == MAX_TAIL_CALL_CNT